views/components/form-fields-next/macros.njk

```diff
@@ -50,52 +50,78 @@
+            class="ff-text ff-width-2"
+            name="{{ dateInputName(field, 'month', rangePeriod) }}"
+            type="number"
+            pattern="[0-9]*"  min="1"
+            autocomplete="off"
+            {% if field.isRequired %}required aria-required="true"{% endif %}
+            value="{{ field.value[rangePeriod].month if rangePeriod else field.value.month }}"
+
+        />
+    </label>
+    {% if includeYear %}
         <label class="ff-date__field">
-            <span class="ff-label">{{ __('global.misc.day') }}</span>
-            <input
-                class="ff-text ff-width-2"
-                name="{{ field.name }}[day]"
-                type="number"
-                pattern="[0-9]*"  min="1"
-                autocomplete="off"
-                {% if field.isRequired %}required aria-required="true"{% endif %}
-                value="{{ field.value.day }}"
-            />
-        </label>
-        <label class="ff-date__field">
-            <span class="ff-label">{{ __('global.misc.month') }}</span>
+            <span class="ff-label">{{ __('global.misc.year') }}</span>
             <input
-                class="ff-text ff-width-2"
-                name="{{ field.name }}[month]"
+                class="ff-text ff-width-4"
+                name="{{ dateInputName(field, 'year', rangePeriod) }}"
                 type="number"
-                pattern="[0-9]*"  min="1"
+                pattern="[0-9]*" min="{{ field.settings.minYear | default(1900) }}"
                 autocomplete="off"
+                size="20"
                 {% if field.isRequired %}required aria-required="true"{% endif %}
-                value="{{ field.value.month }}"
+                value="{{ field.value[rangePeriod].year if rangePeriod else field.value.year }}"
+
             />
         </label>
-        {% if includeYear %}
-            <label class="ff-date__field">
-                <span class="ff-label">{{ __('global.misc.year') }}</span>
-                <input
-                    class="ff-text ff-width-4"
-                    name="{{ field.name }}[year]"
-                    type="number"
-                    pattern="[0-9]*" min="{{ field.settings.minYear | default(1900) }}"
-                    autocomplete="off"
-                    size="20"
-                    {% if field.isRequired %}required aria-required="true"{% endif %}
-                    value="{{ field.value.year }}"
-                />
-            </label>
-        {% endif %}
+    {% endif %}
+{% endmacro %}
+
+{% macro inputDate(field, includeYear = true) %}
+    <fieldset class="ff-date" id="field-{{ field.name }}">
+        <legend class="ff-label ff-date__legend">
+            {{ field.label }}
+        </legend>
+        {{ explanationFor(field) }}
+        {{ dayMonthYearInputs(field, includeYear = includeYear) }}
+    </fieldset>
+{% endmacro %}
+
+{% macro inputDateRange(field) %}
+    <fieldset class="ff-date ff-date--range" id="field-{{ field.name }}">
```

Comment:
This needs a bit of thought, not 100% sure about the grouping and semantically the "start" and "end" labels will need something to connect that text with the field (should possibly be two fieldsets). I know this is still a work in progress.

Reply:
Good shout. Made each bit a `<fieldset>` (eg. the overall legend/explanation, the start and the end fields):
![image](https://user-images.githubusercontent.com/394376/58325079-97a90300-7e20-11e9-85e5-bb92c70b638c.png)